core/embed/rust/src/ui/model_mercury/flow/confirm_reset.rs

```diff
@@ -0,0 +1,169 @@
+        }
+    }
+
+    fn handle_event(&'static self, msg: FlowMsg) -> StateChange {
+        match (self, msg) {
+            (Self::Intro, FlowMsg::Info) => Self::Menu.transit(),
+            (Self::Menu, FlowMsg::Cancelled) => Self::Intro.swipe_right(),
+            (Self::Menu, FlowMsg::Choice(0)) => self.return_msg(FlowMsg::Cancelled),
+            (Self::Confirm, FlowMsg::Confirmed) => self.return_msg(FlowMsg::Confirmed),
+            (Self::Confirm, FlowMsg::Info) => Self::Menu.transit(),
+            _ => self.do_nothing(),
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ConfirmResetRecover {
+    Intro,
+    Menu,
+}
+
+impl FlowState for ConfirmResetRecover {
+    #[inline]
+    fn index(&'static self) -> usize {
+        *self as usize
+    }
+
+    fn handle_swipe(&'static self, direction: SwipeDirection) -> StateChange {
+        match (self, direction) {
+            (Self::Intro, SwipeDirection::Left) => Self::Menu.swipe(direction),
+            (Self::Menu, SwipeDirection::Right) => Self::Intro.swipe(direction),
+            (Self::Intro, SwipeDirection::Up) => self.return_msg(FlowMsg::Confirmed),
+            _ => self.do_nothing(),
+        }
+    }
+
+    fn handle_event(&'static self, msg: FlowMsg) -> StateChange {
+        match (self, msg) {
+            (Self::Intro, FlowMsg::Info) => Self::Menu.transit(),
+            (Self::Menu, FlowMsg::Cancelled) => Self::Intro.swipe_right(),
+            (Self::Menu, FlowMsg::Choice(0)) => self.return_msg(FlowMsg::Cancelled),
+            _ => self.do_nothing(),
+        }
+    }
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub extern "C" fn new_confirm_reset(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, new_confirm_reset_obj) }
+}
+
+fn new_confirm_reset_obj(_args: &[Obj], kwargs: &Map) -> Result<Obj, error::Error> {
+    let recovery: bool = kwargs.get_or(Qstr::MP_QSTR_recovery, false)?;
+
+    let (title, br, cancel_btn_text) = if recovery {
+        (
+            TR::recovery__title_recover.into(),
+            ButtonRequestCode::ProtectCall.with_name("recover_device"),
+            TR::recovery__title_cancel_recovery.into(),
+        )
+    } else {
+        (
+            TR::reset__title_create_wallet.into(),
+            ButtonRequestCode::ResetDevice.with_name("setup_device"),
+            // FIXME: TR::reset__cancel_create_wallet should be used but Button text on
+            // multiple lines not supported yet
+            TR::buttons__cancel.into(),
+        )
+    };
+
+    let par_array: [Paragraph<'static>; 3] = [
+        Paragraph::new(&theme::TEXT_MAIN_GREY_LIGHT, TR::reset__by_continuing)
+            .with_bottom_padding(17),
+        Paragraph::new(&theme::TEXT_SUB_GREY, TR::reset__more_info_at),
+        Paragraph::new(&theme::TEXT_SUB_GREY_LIGHT, TR::reset__tos_link),
+    ];
+    let paragraphs = Paragraphs::new(par_array);
```

Comment:
```suggestion
    let paragraphs = ParagraphVecShort::from_iter([
        Paragraph::new(&theme::TEXT_MAIN_GREY_LIGHT, TR::reset__by_continuing)
            .with_bottom_padding(17),
        Paragraph::new(&theme::TEXT_SUB_GREY, TR::reset__more_info_at),
        Paragraph::new(&theme::TEXT_SUB_GREY_LIGHT, TR::reset__tos_link),
    ])
    .into_paragraphs();
```

Nit: this seems to save another 1K.

We should probably delete `impl<'a, const N: usize> ParagraphSource<'a> for [Paragraph<'a>; N]` and use ParagraphVecShort/ParagraphVecLong everywhere but that's out of scope.

Reply:
Thank you. 

Since you approved already, I directly rebased with your suggested changes.
I will merge after CI.